README.md

```diff
@@ -8,33 +8,103 @@ This crate provides a Rust implementation of the Schnorr signature scheme for th
 ## About
 The Schnorr signature scheme, named after its creator Claus Schnorr, is a digital signature scheme renowned for its simplicity. The scheme provides a simple method of creating short signatures. 
 
-The implementation has been created using the
-Poseidon hash function, the paper for which can
-be found [here](https://eprint.iacr.org/2019/458.pdf).
+The implementation has been created using the [`jubjub`](https://github.com/dusk-network/jubjub) elliptic curve and the [`Poseidon`](https://github.com/dusk-network/Poseidon252) hash function, the paper for which can be found [here](https://eprint.iacr.org/2019/458.pdf).
 
-For a reference to the algorithm, please see the 
-[docs](https://app.gitbook.com/@dusk-network/s/specs/proposals/rfc/rfc-17).
+The signature scheme is implemented within the [Phoenix](https://github.com/dusk-network/phoenix-core/blob/master/docs/protocol-description.pdf) transaction model and is based on the Schnorr Sigma protocol, compiled alongside the Fiat–Shamir transformation, to serve as a non-interactive signature scheme. Specifically, the Phoenix protocol employs a variant that utilizes double Schnorr signatures, verifiable with double public keys, enabling the delegation of computational processes within the protocol's later stages.
 
 ## Library Structure
-The library is partitioned into two components:
+The library is partitioned into the following components:
 
-- **Keys**: Module containing the secret key structure for signing messages, and the public key & keypair structures used in verification.
-- **Signatures**: Module containing functions to verify the validity of Schnorr signatures.
-- **Gadgets**: Contains the Plonk gadgets for signature verification.
+- **Keys**: Module containing the secret key structure for signing messages, as well as the public key and the double public key structures used in verification.
+- **Signatures**: Module containing the standard and double signature structs as well as functions to verify the validity of Schnorr signatures and double Schnorr signatures.
+- **Gadgets**: Contains the Plonk gadgets for in-circuit verification of Schnorr signatures and double Schnorr Signatures.
 
-## Schnorr Sigma Protocol
+## Signature Scheme Description
 
-The signature scheme implemented within the Phoenix protocol is based on the Schnorr Sigma protocol, compiled alongside the Fiat–Shamir transformation, to serve as a non-interactive signature scheme. Specifically, the Phoenix protocol employs a variant that utilizes double keys, enabling the delegation of computational processes within the protocol's later stages.
+### Notation
 
-### Signature Scheme Description
+In the following:
+- Multiplication of a point $P$ by a scalar $s$ stands for adding $P$ $s$-times to itself.
+- $\mathbb{Z}_q$ the set of congruence classes modulo $q$
```

Comment:
~~I would add "$\mathbb{Z}_q$ **is** the set [...]" + missing final dot in the sentence.~~

In reviewing the part below, I realized that in general we do not want the group $\mathbb{Z}_q$, but the field $\mathbb{F}_q$. Although the two things as sets have the same elements, in a group we only consider the addition of elements, and in a field both the addition and the product operations are defined. It is a technical detail only, but since in the protocol we need the field, I would define it here:
- $\mathbb{F}_q$ is the prime finite field of order $q$